apps/pipelines/nodes/nodes.py

```diff
@@ -62,32 +61,52 @@ class RenderTemplate(PipelineNode):
             label="Render a template", documentation_link=settings.DOCUMENTATION_LINKS["node_template"]
         )
     )
-
     template_string: str = Field(
         description="Use {{your_variable_name}} to refer to designate input",
         json_schema_extra=UiSchema(widget=Widgets.expandable_text),
     )
 
-    def _process(self, input, node_id: str, **kwargs) -> PipelineState:
-        def all_variables(in_):
-            return {var: in_ for var in meta.find_undeclared_variables(env.parse(self.template_string))}
-
+    def _process(self, input, node_id: str, state: PipelineState, **kwargs) -> PipelineState:
         env = SandboxedEnvironment()
         try:
-            if isinstance(input, BaseMessage):
-                content = json.loads(input.content)
-            elif isinstance(input, dict):
-                content = input
-            else:
-                content = json.loads(input)
-                if not isinstance(content, dict):
-                    # e.g. it was just a string or an int
-                    content = all_variables(input)
-        except json.JSONDecodeError:
-            # As a last resort, just set the all the variables in the template to the input
-            content = all_variables(input)
-        template = SandboxedEnvironment().from_string(self.template_string)
-        output = template.render(content)
+            content = {
+                "input": input,
+                "messages": state.get("messages", []),
+                "temp_state": state.get("temp_state", {}),
+            }
+
+            if state.get("messages"):
+                for message in state["messages"]:
+                    if isinstance(message, dict):
+                        content.update(message)
+                    else:
+                        content["input"] = message
```

Comment:
I'm not following what this is for. Can you explain.

Reply:
OK, but why are we adding all the messages directly into the content dict? It's going to continually overwrite the previous values. I also don't think it's useful to have `state.messages` in the template context.